site/source/docs/porting/exceptions.rst

```diff
@@ -113,11 +113,51 @@ property. For example:
 Stack traces within Wasm code are not supported in :ref:`JavaScipt-based
 exceptions <javascript-based-exception-support>`.
 
-Exception Messages
-------------------
+
+.. _handling-c-exceptions-from-javascript:
+
+Handling C++ exceptions from JavaScript
+---------------------------------------
 
 You can also catch and examine the type and the message of C++ exceptions from
-JavaScript. See :ref:`handling-c-exceptions-from-javascript`.
+JavaScript, in case they inherit from ``std::exception`` and thus have ``what``
+method.
+
+``getExceptionMessage`` returns a list of two strings: ``[type, message]``. the
+``message`` is the result of calling ``what`` method in case the exception is a
+subclass of ``std::exception``. Otherwise it will be just an empty string.
+
+.. code-block:: javascript
+
+  try {
+    ... // some code that calls WebAssembly
+  } catch (e) {
+    console.log(getExceptionMessage(e).toString());
+  } finally {
+    ...
+  }
+
+In case the thrown value is an integer 3, this will print ``int,``, because the
```

Comment:
Hmm.. will it really not print 3 somewhere?  That seems like it would be useful to have.

Reply:
I don't think C++ itself provides printing functions other than `what()` (which you can override for your custom classes)... And the type and `what` message are what get printed (in native platforms) when a thrown exception crashes a program too. Maybe we can implement a custom function that prints the value in case it is a primitive type someday...

Reply:
Couldn't it print the pointer value?  Would that not be 0x3 in this case?

Reply:
No, it is not a pointer that has a value of 3, but rather an exception pointer to a number with a value of 3. Handling the conversion of all non-std::exception pointers to a specific string is not a simple task.

Reply:
It's different in Emscripten EH and Wasm EH, and in Wasm EH, the thrown value (extracted from the `WebAssembly.Exception` object) is not the value 3, but an address to the `unwindHeader` field of an exception header struct. Some explanation about this is in the PR description of #17064. So to print the value, we need to get the real address in which the value is stored from the actual thrown value (this is a simple pointer arithmetic.. I think [this](https://github.com/emscripten-core/emscripten/blob/3a3329d8e1358b1fd7ac127edfe5186ba5873fcd/system/lib/libcxxabi/src/cxa_exception_js_utils.cpp#L13-L19) does that), and load a value from that address, and figure out what the primitive type for that is because we need to know the type to correctly print it.